### DISCUSSION TITLE
Bug: no-unused-vars inconsistency

### DISCUSSION THREAD

**Discussion by electriquo** [eslint/eslint#16194] [Q&A]
### Environment

Node version: v16.16.0
npm version: v8.14.0
Local ESLint version: v8.21.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 21.6.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    es2022: true,
  },
  rules: {
    "no-unused-vars": ["error", { argsIgnorePattern: "^_" }],
  }
};

```
</details>

```js
(a) => {};
Object(null, (b, c) => { c.e() });
```

### What did you expect to happen?

```shell
$ npx eslint .

/foo/bar/baz/example.js
  1:2   error  'a' is defined but never used. Allowed unused args must match /^_/u  no-unused-vars
  2:15  error  'b' is defined but never used. Allowed unused args must match /^_/u  no-unused-vars
✖ 2 problem (2 error, 0 warnings)
```

### What actually happened?

```shell
$ npx eslint .

/foo/bar/baz/example.js
  1:2  error  'a' is defined but never used. Allowed unused args must match /^_/u  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I would expect that variable `b` will be caught by eslint in the same way variable `a` was since `b` is not in use.
Is it an intended behavior or is it a bug?

**Comment by aladdin-add** [comment]
it's expected. the rule has an option: `args`.  its default value is `"after-used"`: unused positional arguments that occur before the last used argument will not be checked, but all named arguments and all positional arguments after the last used argument will be checked.

you can set it to`"args": "all"` to report it.


https://eslint.org/docs/latest/rules/no-unused-vars#args
